IEEE-PSH/hackpsh-web
# Clerk Authentication Setup (Part 1) - Pre-onboarding

# Description

* Allows for new user(s) to sign up, and get redirected to /onboarding.
* Added attribute: `onboardingComplete` to Clerk Session Claims
* Bumped all project dependencies
* Created Stubs for /onboarding and /dashboard
* Added Login / Register Button(s)


# Test Guidelines

1. Clone Repository and Switch to this Branch
1. Create a file in the project root folder called `.env` - This will hold our application(s) secret(s).
1. Retrieve the contents for this file via [Development Channel](https://discord.com/channels/621377433194725376/1148099579338821702/1159898763184316527)
1. Start & Run the Project Locally & Go To `http://localhost`
1. Click the Register Button and Sign Up Via Google
1. Evaluate if you are at the `/onboarding` page and you see a UserProfile button with your account details.
1. Manually go to `http://localhost:3000/dashboard` and see if you are redirected to `/onboarding`
1. Log Out & Evaluate if you are returned to `/` route (Home Page
1. Click the Login Button and Sign In Via Google
Evaluate if you are at the `/onboarding` page and you see a UserProfile button with your account details.
1. Manually go to `http://localhost:3000/dashboard` and see if you are redirected to `/onboarding`
1. Log Out & Evaluate if you are returned to `/` route (Home Page
